### PR TITLE
Adding perfmon test to avocado

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Signed-off-by: praveen@linux.vnet.ibm.com
added perfmon Test case

```
v1: use git for perfmon test 
v1: build (make)
v1:run test
v1: validate output if failure occur notify user about test fail 
```

### CODE REVIEW

File: perf/perfmon.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    """
+
+    def setUp(self):
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        smm = SoftwareManager()
+
+        for package in ("gcc", "make"):
+            if not smm.check_installed(package) and not smm.install(package):
```
Comment (apahim) [inline perf/perfmon.py:39]: This has to be an `or` instead of an `and`.

File: perf/perfmon.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    """
+    performance monitoring on Linux : test  perf_events on Linux
+    """
+
+    def setUp(self):
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        smm = SoftwareManager()
```
Comment (apahim) [inline perf/perfmon.py:36]: Dup

Comment (PraveenPenguin) [inline perf/perfmon.py:36]: thanks into looking into this  make change and open a new PR

File: perf/perfmon.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        os.chdir(self.srcdir)
+
+        build.make('.')
+
+    def test(self):
+
+        out = process.system_output('%s ' % os.path.join(
+            self.srcdir, 'tests/validate'))
+        if 'fail' in out:
+            self.error("test failed:check manually")
```
Comment (apahim) [inline perf/perfmon.py:55]: `self.fail()` is the best fit to say the test failed (is opposition to the test `PASS` ).

Comment (PraveenPenguin) [inline perf/perfmon.py:55]: sure will make change and open a new PR

Regards
Praveen

**Comment by PraveenPenguin** [comment]
@apahim  raised new PR #126 with changes Please have a look.

Thanks

**Pull request closed without merging**